parlai/agents/transformer/biencoder.py

```diff
@@ -21,6 +21,13 @@ def vectorize(self, *args, **kwargs):
         obs = TorchRankerAgent.vectorize(self, *args, **kwargs)
         return obs
 
+    def _vectorize_text(self, *args, **kwargs):
+        """override to add start end tokens. necessary for fixed cands"""
+        if 'add_start' in kwargs:
+            kwargs['add_start'] = True
+            kwargs['add_end'] = True
+        return super()._vectorize_text(*args, **kwargs)
```

Comment:
a few questions:
1) do we still need these `_set_text_vec` and `vectorize` overrides? 
2) why do we have this check here for `if 'add_start' in kwargs`?
3) we may experience the problem where the observation is cached and we continually add start/end tokesn to the labels at each epoch. how do we get around this? the way i handled it for the text case is to check for the presence of an `added_start_end_tokens` key in the `_set_text_vec` function (see below). this probably requires some careful thought/may be cleaner with a refactor. cc @stephenroller.

Reply:
1. I *think* so - this relates to the issue where the fixed candidate labels are vectorized directly via the `_vectorize_text` function rather than the `vectorize` function. 
2. Sometimes, this function is called with `add_start` as a positional arg, rather than a keyword arg; i am overriding this specifically for the case of building fixed cands, which calls this with `_vectorize_text(..., add_start=...)`, so i check for that case. Also, if i do add it in when it's called as a positional arg, I'll get an error where the function is passed two values for `add_start`
3. Yeah, this is actually handled below in the `_set_text_vec` function. Ideally it is handled in the `_set_label_vec` function in the torch agent as well.

Reply:
re: 3 -- after looking into it a bit more, ti should be handled in set_label_vec because we do not call _vectorize_text again if the label_vec is inside the obs

Reply:
so sorry, i'm junior, how to download the whole of these models to replace the old ones in my pc, cause the aren't work properly